src/lib/lib_duplicity.py

```diff
@@ -185,10 +197,16 @@ def _render_deletion_cron(self):
             deletion_frequency = "40 * * * *"
         elif deletion_frequency == "daily":
             deletion_frequency = "0 23 * * *"
+        is_juju3 = subprocess.getoutput("ls /usr/bin/ | grep juju-exec")
+        if is_juju3:
+            juju_binary = "juju-exec"
+        else:
+            juju_binary = "juju-run"
```

Comment:
**nitpick (non-blocking):** 

`juju_binary` is not precise.

Maybe `juju_exec_binary`.

Reply:
i will add this functionality in another PR